### ISSUE TITLE
modularization

### ISSUE THREAD

**Issue by sheepdreamofandroids** [google/rejoiner#92]
I'm feeling that rejoiner could be split into some modules which would make it more generally useful and easier to adapt to different usecases. I'm thinking out loud here so please tell me I'm wrong before I start on a PR ;-)

1. protobuf -> graphql schema conversion
2. graphql schema modification (now depends on protobuf)
3. annotations for 2
4. binding with guice

Splitting these into maven modules would allow to add these for example:

5. swagger -> graphql schema conversion
6. binding with spring

I think the hardest work would be to make the modification part independent of protobuf without losing efficiency.

cheers

**Comment by siderakis** [comment]
Hey, we are starting work on the multi-module branch. First step is extracting out rejoined-guice from the core. PRs on that brach are welcome :-)

**Comment by sheepdreamofandroids** [comment]
Hey that's great! It was me threatening with a PR that made you do it, right?   ;-)
I'd love to get involved but I'm not sure how to do that without getting into huge merge conflicts.

BTW, I saw that the new module is using gradle. Do you want to move everything to gradle?

**Comment by siderakis** [comment]
Yes! also my coworker had a need for things to be modularized too.

I'm thinking of moving everything to maven, gradle would be fine also but supporting one fully is the goal (right now it's a mix).
